components/TxForm.tsx

```
@@ -37,94 +56,138 @@ export const TxForm = () => {
+      <form onSubmit={onSubmit}>
+        <FormControl isInvalid={Boolean(errors.recipient?.message)}>
+          <FormLabel>Recipient address</FormLabel>
           <Input
-            type="number"
             variant="outline"
             size="lg"
             pr="4.5rem"
-            height="100%"
-            value={tokenAmount}
-            onChange={(e) => setTokenAmount(e.target.value)}
+            height="4rem"
+            mb=".25rem"
+            {...register('recipient', {
+              required: 'This is required',
+              validate: (value) => validateRailgunAddress(value) || 'Invalid railgun address',
+            })}
+          />
+          <FormErrorMessage my=".25rem">
+            {errors.recipient && errors.recipient.message}
+          </FormErrorMessage>
+        </FormControl>
+        <FormControl isInvalid={Boolean(errors.token?.message)} mt=".5rem">
+          <FormLabel>Token</FormLabel>
+          <TokenInput
+            {...register('token')}
+            onSelect={(token) => {
+              setValue('token', token.name);
+              setSelectedToken(token);
+            }}
+          />
+        </FormControl>
+        <FormControl isInvalid={Boolean(errors.amount?.message)}>
+          <FormLabel>Amount</FormLabel>
+          <InputGroup size="lg" width="auto" height="4rem">
+            <Input
+              variant="outline"
+              size="lg"
+              pr="4.5rem"
+              height="100%"
+              {...register('amount', {
+                required: 'This is required',
+                onChange: (e) => {
+                  if (e.target.value && !isNaN(e.target.value)) {
+                    setTokenAmount(e.target.value);
+                  }
+                },
+                validate: (value) => {
+                  try {
+                    return (
+                      Boolean(
+                        parseUnits(value || '0', selectedToken?.decimals).gte(BigNumber.from('0'))
```

Is there a reason to allow shielding with a value of 0?
```suggestion
                        parseUnits(value || '0', selectedToken?.decimals).gt(BigNumber.from('0'))
```

---

Good question, I don't think so, we can surface a different error message for a 0